atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/DefaultTransactionKeyValueServiceManager.java

```diff
@@ -42,6 +45,16 @@ public Optional<KeyValueService> getKeyValueService() {
         return delegate;
     }
 
+    @Override
+    public DdlManager getDdlManager() {
+        return this;
+    }
+
+    @Override
+    public void createTables(Map<TableReference, byte[]> tableRefToTableMetadata) {
+        delegate.get().createTables(tableRefToTableMetadata);
+    }
```

Comment:
So I think we absolutely want to keep TransactionKeyValueServiceManager as like the central piece that you can get smaller pieces from. So DdlManager being a separate interface makes sense to me, because, at least in my current view, this is the only interface I'd want eventually exposed on TransactionManager.

That said, the DefaultTransactionKeyValueServiceManager I just implemented both for convenience, as in this implementation, it's just a call to the delegate KeyValueService.

I'll follow your advice and just split the implementation up for SingleKvsDdlManager, and I'll rename DefaultTransactionKeyValueServiceManager to SingleTransactionKeyValueServiceManager to match.

Reply:
Actually I'll go for Delegating* prefix, single feels weird.

Reply:
Done